salt/minion.py

```diff
@@ -1364,11 +1364,41 @@ def connect_master(self, failed=False):
         """
         Return a future which will complete when you are connected to a master
         """
+        # Consider refactoring so that eval_master does not have a subtle side-effect on the contents of the opts array
```

Comment:
Someday, `opts` (`__opts__`, `__salt__`, *et al*) will be readonly, rite?

Reply:
I'm not sure how quickly that could happen. In any case, we want to tamper with `__opts__` as little as possible.